assets/src/edit-story/components/form/color/colorInput.js

```diff
@@ -0,0 +1,240 @@
+`;
+
+const ColorPreview = styled.div`
+  ${colorStyles}
+  top: 6px;
+  left: 6px;
+  padding: 0;
+  background: transparent;
+  cursor: pointer;
+`;
+
+const ColorPreviewButton = styled(ColorPreview).attrs(buttonAttrs)`
+  ${buttonStyle}
+  padding: 0;
+  border: none;
+`;
+
+const ColorPreviewInsideButton = styled(ColorPreview)`
+  border: none;
+  transform: translate(-1px, -1px);
+`;
+
+const CurrentColor = styled.div`
+  ${colorStyles}
+`;
+
+const Transparent = styled.div`
+  ${colorStyles}
+
+  background-image: conic-gradient(
+    #fff 0.25turn,
+    #d3d4d4 0turn 0.5turn,
+    #fff 0turn 0.75turn,
+    #d3d4d4 0turn 1turn
+  );
+  background-size: 66.67% 66.67%;
+`;
+
+const TextualPreview = styled.div`
+  padding: 6px 12px 6px 38px;
+  text-align: left;
+  flex-grow: 1;
+  height: 32px;
+`;
+
+function ColorInput({
+  onChange,
+  hasGradient,
+  hasOpacity,
+  value,
+  label,
+  colorPickerActions,
+  changedStyle,
+}) {
+  const isMixed = value === MULTIPLE_VALUE;
+  value = isMixed ? '' : value;
+
+  const previewStyle = getPreviewStyle(value);
+  const previewText = getPreviewText(value);
+
+  const [pickerOpen, setPickerOpen] = useState(false);
+  const previewRef = useRef(null);
+
+  const {
+    refs: { inspector },
+  } = useInspector();
+
+  const colorType = value?.type;
+  // Allow editing always in case of solid color of if color type is missing (mixed)
+  const isEditable = !colorType || colorType === 'solid';
+
+  const buttonProps = {
+    onClick: () => setPickerOpen(true),
+    'aria-label': label,
+  };
+
+  // Always hide color picker on unmount - note the double arrows
+  useUnmount(() => () => setPickerOpen(false));
+
+  const onClose = useCallback(() => setPickerOpen(false), []);
+  const spacing = useMemo(() => ({ x: 20 }), []);
+
+  return (
+    <>
+      {isEditable ? (
+        // If editable, only the visual preview component is a button
+        // And the text is an input field
+        <Preview ref={previewRef}>
+          <Input
+            aria-label={label}
+            value={value}
+            onChange={onChange}
+            placeholder={isMixed ? MULTIPLE_DISPLAY_VALUE : ''}
+          />
+          <ColorPreviewButton
+            {...buttonProps}
+            color={previewStyle?.backgroundColor}
+          >
+            {(value?.a < 1 || isMixed) && <Transparent />}
+            <CurrentColor role="status" style={previewStyle} />
+          </ColorPreviewButton>
+        </Preview>
+      ) : (
+        // If not editable, the whole component is a button
+        <ColorButton ref={previewRef} {...buttonProps}>
+          <ColorPreviewInsideButton>
+            <Transparent />
+            <CurrentColor role="status" style={previewStyle} />
+          </ColorPreviewInsideButton>
+          <TextualPreview>
+            <Text size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+              {previewText}
+            </Text>
+          </TextualPreview>
+        </ColorButton>
```

Comment:
This button should probably use the same focus display and the input does - so a full rounded blue rectangle around the entire button (and not like the current slight highlight of the border):
![Screen Shot 2021-02-24 at 12 50 31](https://user-images.githubusercontent.com/637548/109043841-bdbba100-769f-11eb-937e-30009ec01e9f.png)

Reply:
Added the blue rectangle, I noticed, however, that the inputs probably need some adjustments for the focus.
It looks like instead of having just a blue rectangle it also has a dark rectangle (primary color). This is done by applying 2 box shadows (added the same for the other button for now) -- it seems to assume that the background of the input is on the "primary" color to leave the transparent impression but the design panel is not using the primary color as the background. (Zoom in more to see it better)

<img width="418" alt="Screenshot 2021-02-25 at 13 36 19" src="https://user-images.githubusercontent.com/3294597/109148386-7f3ce980-7766-11eb-96bc-873a7b8f5678.png">

Or is the two-colored focus border expected? Thoughts?